### PR TITLE
[tune] Make `metrics` parameter optional in pytorch lightning integration

### LINKED ISSUE #11344: [tune] turn metrics into an optional kwarg in PTL TuneReportCallback
### Describe your feature request
[from the docs](https://docs.ray.io/en/master/tune/tutorials/tune-pytorch-lightning.html#codecell5:~:text=This%20callback%20will%20take%20the%20avg_val_loss,as%20the%20loss%20and%20mean_accuracy%2C%20respectively.):
> ```python
> from ray.tune.integration.pytorch_lightning import TuneReportCallback
> callback = TuneReportCallback({
>     "loss": "avg_val_loss",
>     "mean_accuracy": "avg_val_accuracy"
> }, on="validation_end")
> ```
> This callback will take the `avg_val_loss` and `avg_val_accuracy` values from the PyTorch Lightning trainer and report them to Tune as the loss and mean_accuracy, respectively.

I think it makes sense to make `metrics` optional. If the metrics dict isn't passed, the callback should just report all of PTL's metrics to tune with the same names. 

---

CC @richardliaw 

edit: this functionality exists for XGBoost, so I think it makes sense to do this for PTL too

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

PyTorch Lightning callbacks should also work without explicitly specifying metrics. Our other callback integrations already support this mode.

## Related issue number

Closes #11344

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/integration/pytorch_lightning.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ class TuneReportCallback(TuneCallback):
     """
 
     def __init__(self,
-                 metrics: Union[str, List[str], Dict[str, str]],
+                 metrics: Union[None, str, List[str], Dict[str, str]] = None,
```
Comment (sumanthratna) [inline python/ray/tune/integration/pytorch_lightning.py:165]: It's been a really long time since I've used python type checking but shouldn't we do this?
```suggestion
                 metrics: Optional[Union[str, List[str], Dict[str, str]]] = None,
```
I'm pretty sure they're exactly the same, but Optional seems nicer

File: python/ray/tune/integration/pytorch_lightning.py [hunk 2]
```diff
@@ -253,7 +259,7 @@ class TuneReportCheckpointCallback(TuneCallback):
...
```
Comment (sumanthratna) [inline python/ray/tune/integration/pytorch_lightning.py:262]: see above
```suggestion
                 metrics: Optional[Union[str, List[str], Dict[str, str]]] = None,
```

**Review by sumanthratna — APPROVED** [review]
wow...this was really quick! Thanks, @krfricke! Looks great; I left two nit comments.

File: python/ray/tune/integration/pytorch_lightning.py (hunk 1 above)
Comment (krfricke) [reply python/ray/tune/integration/pytorch_lightning.py:165]: Hm, so the nice thing about keeping it the way it is is that it is a bit easier to read (because we don't introduce another bracket). Also, in this particular case we would have to line-break, as otherwise the linter will complain. So for readability, in this case, I'd like to keep it as is.
